### PR TITLE
interfaces: Describe and follow some code conventions

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR is part of the [process separation project](https://github.com/bitcoin/bitcoin/projects/10).

This PR doesn't change behavior at all, it just cleans up code in [`src/interfaces`](https://github.com/bitcoin/bitcoin/tree/master/src/interfaces) to simplify #10102, and [documents](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-conv/doc/developer-notes.md#internal-interface-guidelines) coding conventions there better

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by practicalswift** [comment]
Concept ACK

On behalf of the eternal stream of future generations of Bitcoin Core developers: thank you for cleaning up the code base :)

**Review by promag — COMMENTED** [review]
Code review ACK 2427e93f5823cffcf1158d1cd389b9d2ef703521.

**Comment by ryanofsky** [comment]
Rebased 2427e93f5823cffcf1158d1cd389b9d2ef703521 -> 60ae4c0756ed14eed31a13c3813e9714cab060d0 ([`pr/ipc-conv.3`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.3) -> [`pr/ipc-conv.4`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.4), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-conv.3-rebase..pr/ipc-conv.4)) due to conflict with #18241

File: src/interfaces/chain.h [hunk 1]
```diff
@@ -154,7 +154,10 @@ class Chain
     //! Transaction is added to memory pool, if the transaction fee is below the
     //! amount specified by max_tx_fee, and broadcast to all peers if relay is set to true.
     //! Return false if the transaction could not be added due to the fee or for another reason.
-    virtual bool broadcastTransaction(const CTransactionRef& tx, std::string& err_string, const CAmount& max_tx_fee, bool relay) = 0;
+    virtual bool broadcastTransaction(const CTransactionRef& tx,
+        const CAmount& max_tx_fee,
+        bool relay,
+        std::string& err_string) = 0;
```
Comment (hebasto) [inline src/interfaces/chain.h]: nit: is this needed to break the line here while keeping this function definition one-liner in chain.cpp?

File: src/interfaces/node.h [hunk 2]
```diff
@@ -204,7 +204,7 @@ class Node
     virtual std::unique_ptr<Wallet> loadWallet(const std::string& name, std::string& error, std::vector<std::string>& warnings) = 0;
 
     //! Create a wallet from file
-    virtual WalletCreationStatus createWallet(const SecureString& passphrase, uint64_t wallet_creation_flags, const std::string& name, std::string& error, std::vector<std::string>& warnings, std::unique_ptr<Wallet>& result) = 0;
+    virtual std::unique_ptr<Wallet> createWallet(const SecureString& passphrase, uint64_t wallet_creation_flags, const std::string& name, std::string& error, std::vector<std::string>& warnings, WalletCreationStatus& status) = 0;
```
Comment (hebasto) [inline src/interfaces/node.h]: nit: considering a new signature of the refactored function could a more appropriate semantically name be chosen?
`newWallet` or smth similar?

**Review by hebasto — APPROVED** [review]
ACK 60ae4c0756ed14eed31a13c3813e9714cab060d0.

It seems AppVeyor fail is unrelated to this PR changes.

File: src/interfaces/chain.h (hunk 1 above)
Comment (ryanofsky) [reply src/interfaces/chain.h]: re: https://github.com/bitcoin/bitcoin/pull/18278#discussion_r389315317

> nit: is this needed to break the line here while keeping this function definition one-liner in chain.cpp?

added same wrapping

File: src/interfaces/node.h (hunk 2 above)
Comment (ryanofsky) [reply src/interfaces/node.h]: re: https://github.com/bitcoin/bitcoin/pull/18278#discussion_r389316112

> nit: considering a new signature of the refactored function could a more appropriate semantically name be chosen?
> `newWallet` or smth similar?

Unclear why that name is more semantically appropriate, but not knowing this, it seems reasonable for the interface wrapper method name to have the same name as the function it is wrapping

**Review by ryanofsky — COMMENTED** [review]
Thanks for review

Updated 60ae4c0756ed14eed31a13c3813e9714cab060d0 -> 43c3b391feeb0ad1b2faa4e269b64175399ef5df ([`pr/ipc-conv.4`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.4) -> [`pr/ipc-conv.5`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.5), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-conv.4..pr/ipc-conv.5)) adding suggested wrapping

**Comment by ryanofsky** [comment]
Rebased 43c3b391feeb0ad1b2faa4e269b64175399ef5df -> 25f1a842929dd73bcec26c5ce213adfc65b9cb2e ([`pr/ipc-conv.5`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.5) -> [`pr/ipc-conv.6`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.6), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-conv.5-rebase..pr/ipc-conv.6)) due to confilct with #18115

**Review by hebasto — APPROVED** [review]
re-ACK 25f1a842929dd73bcec26c5ce213adfc65b9cb2e

**Comment by ryanofsky** [comment]
Rebased 25f1a842929dd73bcec26c5ce213adfc65b9cb2e -> 3dc27a15242a22b5301904375e5880372e9b7f4d ([`pr/ipc-conv.6`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.6) -> [`pr/ipc-conv.7`](https://github.com/ryanofsky/bitcoin/commits/pr/ipc-conv.7), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/ipc-conv.6-rebase..pr/ipc-conv.7)) due to conflict with #17477

**Comment by hebasto** [comment]
re-ACK 3dc27a15242a22b5301904375e5880372e9b7f4d, the only change since the [previous](https://github.com/bitcoin/bitcoin/pull/18278#pullrequestreview-372582146) review is rebasing.

**Comment by MarcoFalke** [comment]
ACK 3dc27a15242a22b5301904375e5880372e9b7f4d 🕍

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 3dc27a15242a22b5301904375e5880372e9b7f4d 🕍
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgv6Qv+N5ucWCeIaDagqqWUUjRwyJrTq8kr9yZWzQIsZ9zBbrnaxsr106m6Gc8T
NEmKSbDdgLK+oXFFDRYmO7kvVib0hUoVcfZfq42vSEfqg5mMBKUG6DFAGseKh0Fg
LdsvKuczDSeb/L52uIrdQe/ptzpjCqJvHHiScn7SoOidAiyjFWwdhOtBwV7oRuF9
UgGD10K5O7hy70rmn9JOG7GPI1T5NYG2ZgAy7w2n2ChBNeNv9mSse/pDKOgrbhS8
97Fsl5P+SQq2Fi3aQDLBRd5/sj2nu0BpoKCoowJsRHE4GOeAngdZI6f1EOpYLamH
s91MQOTXMgN3VFvhnKSNSVesNmMbvGs4aCyLSXQvPh4ni7jpwjpPf7czl3IItn0L
KnrgUg7uVquxU2FXOvRaVmKo1/PvicrhWYp2Vh8QN06yi91ALoqUqiWHBkdKiwN6
KoAfUKas0Xn+xgg4h+RKNYh5SFLjNEzResi9+ED5KfDW+sjFR9vzElOHiEf/u+Lb
F661+ybw
=ipNG
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `a78b9d76a84ed28d935046f7fd167bda964ff54cc7339f5435a9c7c3e22c90ed  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108a78b9d76a84ed28d935046f7fd167bda964ff54cc7339f5435a9c7c3e22c90edf010b44f9f2b1844821b370fa2053a03a13908fff0108415e5c6f8a3a899449a55f30838fd0108f1045e791df7f008badf010468d074650083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0109a0979c46b7c5f11349a6e520ca8cc2108f1045e791df7f00814e0115ed627a56b0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010b375ea352a649a0fbd84c64832d3f94a08f1045e791df7f008e5686a90e3bd40020083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010c2f7a04ba21252efc59bf7142c77802808f1045e791df7f008dd227fafd41849030083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Pull request merged**